src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -985,6 +986,120 @@ ExitCode ServerRequests::getUserFromSyncDbId(int syncDbId, User &user) {
+    }
+
+    std::function<bool(int)> progressCallbackArchivingWrapper = [&safeProgressCallback](int percent) {
+        return safeProgressCallback('A', percent);
+    };
+
+
+    SyncPath archivePath;
+    exitCode = LogArchiver::generateLogsSupportArchive(includeArchivedLog, logUploadTempFolder, progressCallbackArchivingWrapper,
+                                                       archivePath, exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in LogArchiver::generateLogsSupportArchive: " << exitCode << " : " << exitCause);
+        IoHelper::deleteDirectory(logUploadTempFolder, ioError);
+        return exitCode;
+    }
+
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, archivePath.string(), found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+    }
+
+    // Upload archive
+    std::function<bool(int)> progressCallbackUploadingWrapper = [&safeProgressCallback](int percent) {
+        return safeProgressCallback('U', percent);
+    };
+
+    for (int i = 0; i < 100; i++) {  // TODO: Remove | Fake progress waiting for the real upload implementation
+        if (!progressCallbackUploadingWrapper(i)) {
+            exitCode = ExitCodeOperationCanceled;
+            break;
+        }
+        Utility::msleep(100);
+    }
+
+
+    // TODO: implement real log upload backend
+
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Error during log upload: " << exitCode << " : " << exitCause);
+        // We do not delete the archive here, The path is stored in the app state and the user can try to upload it manually
+        return exitCode;
+    }
+
+    IoHelper::deleteDirectory(logUploadTempFolder, ioError);  // Delete temp folder if the upload was successful
+
+    long long timestamp =
+        std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()).time_since_epoch().count();
+    std::string uploadDate = std::to_string(timestamp);
+    if (bool found = false; !ParmsDb::instance()->updateAppState(AppStateKey::LastSuccessfulLogUploadeDate, uploadDate, found) ||
+                            !found || !ParmsDb::instance()->updateAppState(AppStateKey::LastLogUploadArchivePath, "", found) ||
+                            !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::updateAppState");
+    }
+    return ExitCodeOk;
+}
+
+ExitCode ServerRequests::cancelLogToSupport(ExitCause &exitCause) {
+    exitCause = ExitCauseUnknown;
+    std::string logUploadStatus;
+    if (bool found = false;
+        !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadStatus, logUploadStatus, found) || !found) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in ParmsDb::getAppState");
+        return ExitCodeDbError;
+    }
+
+    if (logUploadStatus == "C0") {
```

Comment:
Do not harcode strings, create static const varaibles somewhere

Reply:
Will be change with this comment: https://github.com/Infomaniak/desktop-kDrive/pull/117#discussion_r1601570338